tests/storage/benchmarks/conftest.py

```diff
@@ -0,0 +1,168 @@
+import itertools
+import logging
+import os
+import tempfile
+import urllib.request
+from urllib.parse import urlparse
+from uuid import uuid4
+
+import pytest
+
+from lib.commands import SSHCommandFailed
+
+from .helpers import load_results_from_csv, str_to_tuple
+
+MAX_LENGTH = 64 * (1024**3)  # 64GiB
+
+
+# use vhd, qcow2, raw... when image_format support will be available
+@pytest.fixture(scope="module", params=["vdi"])
+def image_format(request):
+    return request.param
+
+
+@pytest.fixture(scope="module")
+def running_unix_vm_with_fio(running_unix_vm):
+    vm = running_unix_vm
+    install_cmds = (
+        ("command -v apt", "apt update && apt install -y fio", "apt remove -y fio"),
+        ("command -v dnf", "dnf install -y fio", "dnf remove -y fio"),
+        ("command -v yum", "yum install -y fio", "yum remove -y fio"),
+        ("command -v apk", "apk add fio", "apk del fio"),
+    )
+
+    for check_cmd, install_cmd, remove in install_cmds:
+        try:
+            vm.ssh(check_cmd, check=True)
+            logging.info(f">> Installing fio with {install_cmd}")
+            vm.ssh(install_cmd, check=True)
+            remove_cmd = remove
+            break
+        except SSHCommandFailed:
+            ...
+    else:
+        raise RuntimeError("Unsupported package manager: could not install fio")
+
+    yield vm
+
+    # teardown
+    logging.info(f"<< Removing fio with {remove_cmd}")
+    vm.ssh(remove_cmd, check=False)
+
+
+@pytest.fixture(scope="module")
+def vdi_on_local_sr(host, local_sr_on_hostA1, image_format):
+    sr = local_sr_on_hostA1
+    vdi = sr.create_vdi("testVDI", MAX_LENGTH)
```

Comment:
The `image_format` has to be given to `create_vdi`
```suggestion
    vdi = sr.create_vdi("testVDI", MAX_LENGTH, image_format=image_format)
```

Reply:
Image format is not supported yet, resulting in an error. I will provide it as a comment though.

Reply:
Fixed in [377d577](https://github.com/xcp-ng/xcp-ng-tests/pull/324/commits/377d577ab02ecc3d705702a2c5c306afca3147a3)